bokeh/core/has_props.py

```diff
@@ -471,86 +404,57 @@ def lookup(cls, name: str, *, raises: bool = True) -> Optional[PropertyDescripto
             PropertyDescriptor : descriptor for property named ``name``
 
         '''
-        resolved_name = cls._property_aliases().get(name, name)
-        attr = getattr(cls, resolved_name, None)
-        if attr is not None:
+        attr = getattr(cls, name, None)
+        if attr is not None or (attr is None and not raises):
             return attr
-        elif not raises:
-            return None
-        else:
-            raise AttributeError(f"{cls.__name__}.{name} property descriptor does not exist")
-
-    @classmethod
-    def properties_with_refs(cls):
-        ''' Collect the names of all properties on this class that also have
-        references.
-
-        This method *always* traverses the class hierarchy and includes
-        properties defined on any parent classes.
-
-        Returns:
-            set[str] : names of properties that have references
-
-        '''
-        return accumulate_from_superclasses(cls, "__properties_with_refs__")
-
-    @classmethod
-    def properties_containers(cls):
-        ''' Collect the names of all container properties on this class.
-
-        This method *always* traverses the class hierarchy and includes
-        properties defined on any parent classes.
-
-        Returns:
-            set[str] : names of container properties
-
-        '''
-        return accumulate_from_superclasses(cls, "__container_props__")
+        raise AttributeError(f"{cls.__name__}.{name} property descriptor does not exist")
 
     @classmethod
+    @lru_cache(None)
     def properties(cls):
         ''' Collect the names of properties on this class.
 
         This method *optionally* traverses the class hierarchy and includes
         properties defined on any parent classes.
 
-        Args:
-            with_bases (bool, optional) :
-                Whether to include properties defined on parent classes in
-                the results. (default: True)
-
         Returns:
            set[str] : property names
 
         '''
-        return accumulate_from_superclasses(cls, "__properties__")
+        props = dict()
+        for c in cls.__mro__:
+            props.update(getattr(c, "__properties__", {}))
```

Comment:
I'm wondering why sometimes you're using `{}` and otherwise `dict()`? I guess it's being influenced by `set()`?